src/store_cache.rs

```diff
@@ -76,4 +78,14 @@ impl StoreCache {
         })
 
     }
+
+    /// Similar to `fetch_url`, but also parses the response as JSON.
+    pub fn fetch_json_url(&self, store: &Store, key: CacheKey, session: &HttpClient, url: &str)
```

Comment:
Since `fetch_url()` is now only used in this method, I'd prefer to merge them. Any counter-arguments?